src/run/headers.ts

```diff
@@ -242,14 +250,27 @@ export const setCacheControlHeaders = (
       .log('NetlifyHeadersHandler.trailingSlashRedirect')
   }
 
-  if (status === 404 && request.url.endsWith('.php')) {
-    // temporary CDN Cache Control handling for bot probes on PHP files
-    // https://linear.app/netlify/issue/FRB-1344/prevent-excessive-ssr-invocations-due-to-404-routes
-    headers.set('cache-control', 'public, max-age=0, must-revalidate')
-    headers.set('netlify-cdn-cache-control', `max-age=31536000, durable`)
+  const cacheControl = headers.get('cache-control')
+  if (status === 404) {
+    if (request.url.endsWith('.php')) {
+      // temporary CDN Cache Control handling for bot probes on PHP files
+      // https://linear.app/netlify/issue/FRB-1344/prevent-excessive-ssr-invocations-due-to-404-routes
+      headers.set('cache-control', 'public, max-age=0, must-revalidate')
+      headers.set('netlify-cdn-cache-control', `max-age=31536000, durable`)
+      return
+    }
+
+    if (
+      process.env.CACHE_404_PAGE &&
+      request.url.endsWith('/404') &&
```

Comment:
I know this has the potential for false positives if a site had a path like, `/products/404`, this feature is behind an env var and only intended for a single customer, so I think it's not worth the effort to make this any stricter